docs/source/core/adapters/components.rst

```diff
@@ -0,0 +1,90 @@
+Adapter Components
+==================
+
+Adapters can be considered as any set of parameters that are added to a pre-existing module/model. In our case, we currently support the standard adapter in literature, more advanced adapter modules are being researched and can potentially be supported by NeMo.
+
+An adapter module can be any pytorch module, but it must follow certain straightforward requirements -
+
+1) The model accepts an input of some input dimension, and its output must match this dimension.
+2) Ideally, the module is initialized such that the output of the adapter when initialized is such that it does not modify the original input. This allows the model to produce the same output results, even when additional parameters have been added.
+
+According to Junxian et al :cite:`adapters-Junxian2021unified`, we can consider an adapter being represented as three components -
+
+1) Functional form - the trainable parameters that will modify the input
+2) Insertion form - Where the adapter outputs are integrated with the original input. The input to the adapters can be the last output of the layer, the input to some attention layer, or even the original input to the module itself (before even the modules forward pass).
+3) Composition function - How the adapters outputs are integrated with the inputs. It can be as simple as residual addition connection, or concatenation, or point-wise multiplication etc.
+
+Functional Form - Adapter Networks
+==================================
+
+Adapter modules represent the functional form of the adapter. We discuss an example of a most commonly used adapter module found in literature, titled the ``LinearAdapter`` (or Houlsby Adapter) :cite:`adapters-houlsby2019adapter`.
+
+.. note::
```

Comment:
Not sure how the formatting will turn up in html pages, but when looking at the rst page in github, it could changed to something like "Note: xxx"

Reply:
This note looks better, it's a type of admonishment commonly used in rtd